### PR TITLE
Fix Mega pokemon getting their base form ability

### PR DESCRIPTION
**Pull request by AustinXII**

### CODE REVIEW

**Comment by AustinXII** [comment]
@Zarel here's a fix but sadly it doesn't account for pokemon with two mega formes, I think it will require some more hardcoding :/

File: js/moveset_import.js [hunk 1]
```diff
@@ -262,7 +260,11 @@ function addSets(pokes) {
 					currentPoke.nameProp = "Custom Set";
 				}
 				currentPoke.isCustomSet = true;
-				currentPoke.ability = getAbility(rows[i + 1].split(":"));
+				if (currentPoke.name.indexOf("-Mega") !== -1) {
+					currentPoke.ability = getAbility(rows[i + 1].split(":"));
+				} else {
```
Comment (AustinXII) [inline js/moveset_import.js]: will probably need something like ``currentPoke.name.indexOf("-Mega-X) {currentPoke.ability = POKEDEX_SM[currentPoke.name].ability;}`` and ``currentPoke.name.indexOf("-Mega-Y") {currentPoke.ability = POKEDEX_SM[currentPoke.name].altAbility;``

File: js/moveset_import.js [hunk 2]
```diff
@@ -262,7 +260,11 @@ function addSets(pokes) {
 					currentPoke.nameProp = "Custom Set";
 				}
 				currentPoke.isCustomSet = true;
-				currentPoke.ability = getAbility(rows[i + 1].split(":"));
+				if (currentPoke.name.indexOf("-Mega") !== -1) {
```
Comment (Zarel) [inline js/moveset_import.js]: It's interesting that you're checking for non-mega first, and then leaving mega for the `else`. I would do it the other way around: the default should be non-mega.

And also, you should probably check that `POKEDEX_SM[currentPoke.name]` and `POKEDEX_SM[currentPoke.name].ability ` even exist.

Comment (AustinXII) [reply js/moveset_import.js]: Honestly I just did this fast so I wouldn’t forget how I did it tomorrow lol. But yes that is probably better, ty!

File: js/moveset_import.js [hunk 3]
```diff
@@ -262,7 +260,12 @@ function addSets(pokes) {
 					currentPoke.nameProp = "Custom Set";
 				}
 				currentPoke.isCustomSet = true;
-				currentPoke.ability = getAbility(rows[i + 1].split(":"));
+				if (currentPoke.name.indexOf("ite") === -1) {
```
Comment (Zarel) [inline js/moveset_import.js]: Err, either `currentPoke.name` is named very badly, or this is wrong and you should be testing item name instead.

Comment (AustinXII) [reply js/moveset_import.js]: Yeah this should be ``currentPoke.item`` but it’s not working as intended so I’m brainstorming a bit more.

**Pull request closed without merging**